book/searcher.js

```diff
@@ -145,6 +145,11 @@ window.search = window.search || {};
             url.push("");
         }
 
+        // encodeURIComponent escapes all chars that could allow an XSS except
+        // for '. Due to that we also manually replace ' with its url-encoded
+        // representation (%27).
+        var searchterms = encodeURIComponent(searchterms.join(" ")).replace(/\'/g, "%27");
+
```

Comment:
no idea what's going on here, maybe i have a newer version of mdbook? happy to remove if we don't want

Reply:
Looks like it. Let's keep it